Source/Core/Core/IOS/USB/Emulated/Skylanders/SkylanderFigure.cpp

```diff
@@ -0,0 +1,462 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "SkylanderFigure.h"
+
+#include <algorithm>
+
+#include <Common/IOFile.h>
+#include <Common/Logging/Log.h>
+#include <Common/Random.h>
+#include <Common/StringUtil.h>
+
+#include <mbedtls/aes.h>
+#include <mbedtls/md5.h>
+
+#include "Skylander.h"
+#include "SkylanderCrypto.h"
+
+namespace IOS::HLE::USB
+{
+u8 PointerToU8(u8* pointer)
+{
+  return *pointer;
+}
+u16 PointerToU16(u8* pointer)
+{
+  return *(u16*)pointer;
+}
+u32 PointerToU32(u8* pointer)
+{
+  return *(u32*)pointer;
+}
+u64 PointerToU64(u8* pointer)
+{
+  return *(u64*)pointer;
+}
+void SkylanderFigure::PopulateSectorTrailers()
+{
+  // Set the sector permissions
+  u32 first_block = 0x690F0F0F;
+  u32 other_blocks = 0x69080F7F;
+  memcpy(&data[0x36], &first_block, sizeof(first_block));
+  for (u32 index = 1; index < 0x10; index++)
+  {
+    memcpy(&data[(index * 0x40) + 0x36], &other_blocks, sizeof(other_blocks));
+  }
+}
+
+void SkylanderFigure::PopulateKeys()
+{
+  for (u8 sector = 0; sector < 0x10; sector++)
+  {
+    u16 keyOffset = (sector * 64) + (3 * 16);
+    u64 key = SkylanderCrypto::CalculateKeyA(sector, data.data());
+
+    for (u8 j = 0; j < 6; j++)
+    {
+      u16 index = keyOffset + (5 - j);
+      u8 byte = (key >> (j * 8)) & 0xFF;
+      data[index] = byte;
+    }
+  }
+}
+
+SkylanderFigure::SkylanderFigure(const std::string& file_path)
+{
+  sky_file = std::move(File::IOFile(file_path, "w+b"));
```

Comment:
`File::IOFile` is already an rvalue here, you don't need to move it.

Reply:
Was an artifact from the other constructor which I copy pasted. Fixed in coming commit